Argo/Types/Decoded/Alternative.swift

```diff
@@ -13,8 +13,15 @@
 infix operator <|> { associativity left precedence 140 }
 
 public func <|> <T>(lhs: Decoded<T>, rhs: Decoded<T>) -> Decoded<T> {
-  if case .Success = lhs {
-    return lhs
+  return lhs.alternative(rhs)
+}
+
+public extension Decoded {
+  func alternative<T>(other: Decoded<T>) -> Decoded<T> {
+    if case .Success = self {
```

Comment:
I'd like to do this without force casting, but on my machines, this removing it (using either the if or the switch implementation) results in this excellent error message:  cannot convert return expression of type 'Decoded<T>' to return type 'Decoded<T>'